alvr/dashboard/src/data_sources.rs

```diff
@@ -260,28 +267,49 @@ impl DataSources {
 
         let ping_thread = thread::spawn({
             let running = Arc::clone(&running);
-            let data_source = Arc::clone(&session_source);
+            let session_source = Arc::clone(&session_source);
             let server_connected = Arc::clone(&server_connected);
             move || {
                 const PING_INTERVAL: Duration = Duration::from_secs(1);
                 let mut deadline = Instant::now();
-                let uri = format!("http://127.0.0.1:{port}/api/ping");
+                let uri = format!("http://127.0.0.1:{port}/api/version");
 
                 let request_agent = ureq::AgentBuilder::new()
                     .timeout_connect(REQUEST_TIMEOUT)
                     .build();
 
                 loop {
-                    let connected = request_agent.get(&uri).call().is_ok();
+                    let maybe_server_version = request_agent
+                        .get(&uri)
+                        .call()
+                        .ok()
+                        .and_then(|r| Version::from_str(&r.into_string().ok()?).ok());
+
+                    let connected = if let Some(version) = maybe_server_version {
+                        // We need exact match because we don't do session extrapolation at the
+                        // dashboard level. In the future we may relax the contraint and consider
+                        // protocol compatibility check for dashboard.
+                        let matches = version == *alvr_common::ALVR_VERSION;
+
+                        if !matches {
+                            error!("Server version mismatch: found {version}. Please remove all previous ALVR installations");
+                        }
+
+                        matches
+                    } else {
+                        false
```

Comment:
This will just fail without an error on old servers. I mean this should never happen in the first place, so it's probably fine, but we could potentially try to do a ping if the version fails. If that succeeds but the version failed then we also know there's a mismatch.

Reply:
I thought about it but it's extra effort. I'll try not to get stuck too much on old code paths